Java/libraries/recognizers-text-datatypes-timex-expression/src/main/java/com/microsoft/recognizers/datatypes/timex/expression/english/TimexConvertEnglish.java

```diff
@@ -86,17 +86,20 @@ public static String convertDate(TimexProperty timex) {
             return TimexConstantsEnglish.DAYS[timex.getDayOfWeek() - 1];
         }
 
-        String month = TimexConstantsEnglish.MONTHS[timex.getMonth() - 1];
         String date = String.valueOf(timex.getDayOfMonth());
 
         String abbreviation = TimexConstantsEnglish.DATE_ABBREVIATION[Integer
                 .parseInt(String.valueOf(date.charAt(date.length() - 1)))];
 
-        if (timex.getYear() != null) {
-            return String.format("%1$s%2$s %3$s %4$s", date, abbreviation, month, timex.getYear()).trim();
+        if (timex.getMonth() != null) {
+            String month = TimexConstantsEnglish.MONTHS[timex.getMonth() - 1];
+            if (timex.getYear() != null)
+            {
+                return String.format(date, abbreviation, month, timex.getYear());
+            }
+            return String.format(date, abbreviation, month);
```

Comment:
```suggestion
            return String.format("%1$s%2$s %3$s", date, abbreviation, month);
```

Following [C#](https://github.com/microsoft/Recognizers-Text/pull/2457/files#diff-246366d7220774e90623ebc0472c32bc5f3cb025976aa994ef12cae0e023b976R110)